Sales Insights

In [1]:
# Importing required libraries and packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns



## DATA LOADING

In [2]:
# Import sheets 
sheets = pd.read_excel("ECOMM DATA.xlsx", sheet_name=None)

/Users/purveshbhave/Desktop/Ecom_Analysis/.venv/lib/python3.13/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)


In [3]:
# Assigning names and creating diffferent data frames form sheets in the xlsl file
df_orders = sheets['Orders']
df_returns = sheets['Returns']
df_people = sheets['People']

df_order = Main transactional data.

df_people = Regional manager data.

df_returns = Returned orders (Contains Order IDs that were returned) 

## DATA UNDERSTANDING 

In [4]:
# Getting Basic overview of all the dataframes 
dfs = [df_orders, df_returns, df_people]

for i, df in enumerate(dfs, start=1):
    print(f"\nDataFrame {i}:")
    print(df.head(5))


DataFrame 1:
   ROW ID         Order ID Order Date  Ship Date     Ship Mode Customer ID  \
0   34323   CA-2012-124891 2012-07-31 2012-07-31      Same Day    RH-19495   
1   26341    IN-2013-77878 2013-02-05 2013-02-07  Second Class    JR-16210   
2   25330    IN-2013-71249 2013-10-17 2013-10-18   First Class    CR-12730   
3   13524  ES-2013-1579342 2013-01-28 2013-01-30   First Class    KM-16375   
4   47221     SG-2013-4320 2013-11-05 2013-11-06      Same Day     RH-9495   

      Customer Name      Segment           City            State  ...  \
0       Rick Hansen     Consumer  New York City         New York  ...   
1     Justin Ritter    Corporate     Wollongong  New South Wales  ...   
2      Craig Reiter     Consumer       Brisbane       Queensland  ...   
3  Katherine Murray  Home Office         Berlin           Berlin  ...   
4       Rick Hansen     Consumer          Dakar            Dakar  ...   

         Product ID    Category  Sub-Category  \
0   TEC-AC-10003033  Technolo

In [5]:
# Getting number of null values form all data frames
for name, df in {
    'Orders': df_orders,
    'Returns': df_returns,
    'People': df_people
}.items():
    print(f"{name}:\n{df.isnull().sum()}\n")

Orders:
ROW ID                0
Order ID              0
Order Date            0
Ship Date             0
Ship Mode             0
Customer ID           0
Customer Name         0
Segment               0
City                  0
State                 0
Country               0
Market                0
Postal Code       41296
Region                0
Product ID            0
Category              0
Sub-Category          0
Product Name          0
Sales                 0
Quantity              0
Discount              0
Shipping Cost         0
Profit                0
Order Priority        0
dtype: int64

Returns:
Returned    0
Order ID    0
Market      0
dtype: int64

People:
Person    0
Region    0
dtype: int64



AS SEEN ABOVE THERE ARE 41296 MISSING VALUES IN ORDERS DATA FRAME IN POSTAL CODE ATTRIBUTE. REST OF THE ATTRIBUTES DONT HAVE ANY NULL/MISSIG VALUES


In [6]:
# Getting info on the datatype of all the attributes present in dataframes
df_orders.dtypes

ROW ID                     int64
Order ID                     str
Order Date        datetime64[us]
Ship Date         datetime64[us]
Ship Mode                    str
Customer ID                  str
Customer Name                str
Segment                      str
City                         str
State                        str
Country                      str
Market                       str
Postal Code              float64
Region                       str
Product ID                   str
Category                     str
Sub-Category                 str
Product Name                 str
Sales                    float64
Quantity                   int64
Discount                 float64
Shipping Cost            float64
Profit                   float64
Order Priority               str
dtype: object

In [7]:
df_people.dtypes

Person    str
Region    str
dtype: object

In [8]:
df_returns.dtypes

Returned    str
Order ID    str
Market      str
dtype: object

AFTER ANALYZING ALL THE ATTRIBUTES IT CAN BE SEEN THAT THE POSTAL CODE FEATURE IN 
df_orders HAS 41296 NULL/ MISSING VALUES AND IS NOT CONTRIBUTING TOWARDS THE ANALYSIS 
AND CAN BE DROPPED.  

## DATA CLEANING

In [9]:
# Dropping the postal code column as it has very high number of missing values
df_orders = df_orders.drop(columns=['Postal Code'])

In [10]:
# Formatting the Order Date and Ship Date columns in proper date time format
df_orders['Order Date'] = pd.to_datetime(df_orders['Order Date'])
df_orders['Ship Date'] = pd.to_datetime(df_orders['Ship Date'])

In [ ]:
# converting yes and no in returned column in df_return to binary input 
df_returns['Returned'] = df_returns['Returned'].apply(lambda x: 1 if x == 'Yes' else 0) # Yes = 1, NO = 0

### Feature Engineering